gax-java/gax/src/test/java/com/google/api/gax/longrunning/OperationTimedPollAlgorithmTest.java

```diff
@@ -0,0 +1,122 @@
+import com.google.api.gax.util.FakeLogHandler;
+import java.util.concurrent.CancellationException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.threeten.bp.Duration;
+
+public class OperationTimedPollAlgorithmTest {
+
+  private static final RetrySettings FAST_RETRY_SETTINGS =
+      RetrySettings.newBuilder()
+          .setInitialRetryDelay(Duration.ofMillis(1L))
+          .setRetryDelayMultiplier(1)
+          .setMaxRetryDelay(Duration.ofMillis(1L))
+          .setInitialRpcTimeout(Duration.ofMillis(1L))
+          .setMaxAttempts(0)
+          .setJittered(false)
+          .setRpcTimeoutMultiplier(1)
+          .setMaxRpcTimeout(Duration.ofMillis(1L))
+          .setTotalTimeout(Duration.ofMillis(5L))
+          .build();
+
+  private FakeLogHandler logHandler;
+
+  @Before
+  public void setUp() {
+    logHandler = new FakeLogHandler();
+    OperationTimedPollAlgorithm.LOGGER.addHandler(logHandler);
+  }
+
+  @After
+  public void tearDown() {
+    OperationTimedPollAlgorithm.LOGGER.removeHandler(logHandler);
+    // redundant null assignment for readability - a new log handler will be used
+    logHandler = null;
+  }
+
+  @Test
+  public void testAlgorithmThatShouldRetry_doesNotLogTimeoutHelpMessage() {
+    FakeApiClock clock = new FakeApiClock(System.nanoTime());
+    OperationTimedPollAlgorithm algorithm =
+        OperationTimedPollAlgorithm.create(FAST_RETRY_SETTINGS, clock);
+    TimedAttemptSettings settings =
+        TimedAttemptSettings.newBuilder()
+            .setGlobalSettings(FAST_RETRY_SETTINGS)
+            .setRetryDelay(Duration.ofMillis(1l))
+            .setRpcTimeout(Duration.ofMillis(1l))
+            .setRandomizedRetryDelay(Duration.ofMillis(1l))
+            .setAttemptCount(0)
+            .setFirstAttemptStartTimeNanos(clock.nanoTime())
+            .build();
+    try {
+      algorithm.shouldRetry(settings);
+    } catch (CancellationException ex) {
+      fail("Unexpected unsuccessful shouldRetry()");
+    }
+    assertTrue(
+        logHandler.getAllMessages().stream()
+            .noneMatch(
+                entry -> entry.contains(OperationTimedPollAlgorithm.LRO_TROUBLESHOOTING_LINK)));
+  }
+
+  @Test
+  public void testAlgorithmThatShouldNotRetry_logsTimeoutHelpMessage() {
+    FakeApiClock clock = new FakeApiClock(System.nanoTime());
+    OperationTimedPollAlgorithm algorithm =
+        OperationTimedPollAlgorithm.create(FAST_RETRY_SETTINGS, clock);
+    TimedAttemptSettings settings =
+        TimedAttemptSettings.newBuilder()
+            .setGlobalSettings(FAST_RETRY_SETTINGS)
+            .setRetryDelay(Duration.ofMillis(1l))
+            .setRpcTimeout(Duration.ofMillis(1l))
+            .setRandomizedRetryDelay(Duration.ofMillis(1l))
+            .setAttemptCount(0)
+            .setFirstAttemptStartTimeNanos(clock.nanoTime())
+            .build();
+    clock.incrementNanoTime(1 * 1000 * 1000 * 1000);
```

Comment:
Consider a quick comment here to clarify what part of the logic this is triggering to force a no-retry.
For example (if this is accurate):
```java
clock.incrementNanoTime(1 * 1000 * 1000 * 1000); // Force Rpc Timeout
```

Reply:
good point. I added this comment